src/Opdex.Platform.Infrastructure/Data/Handlers/Tokens/SelectTokensWithFilterQueryHandler.cs

```diff
@@ -64,18 +64,19 @@ public async Task<IEnumerable<Token>> Handle(SelectTokensWithFilterQuery request
     {
         var sqlParams = new SqlParams(request.MarketId, request.Cursor.Pointer, request.Cursor.Keyword, request.Cursor.Tokens);
 
-        var command = DatabaseQuery.Create(QueryBuilder(request), sqlParams, cancellationToken);
+        var query = DatabaseQuery.Create(QueryBuilder(request), sqlParams, cancellationToken);
 
-        var tokenEntities = await _context.ExecuteQueryAsync<TokenEntity>(command);
+        var tokenEntities = await _context.ExecuteQueryAsync<TokenEntity>(query);
 
         return _mapper.Map<IEnumerable<Token>>(tokenEntities);
     }
 
     private static string QueryBuilder(SelectTokensWithFilterQuery request)
     {
-        var whereFilter = request.MarketId > 0
-            ? $" WHERE ts.{nameof(TokenSummaryEntity.MarketId)} = @{nameof(SqlParams.MarketId)}"
-            : $" WHERE (ts.{nameof(TokenSummaryEntity.MarketId)} IS NULL OR ts.{nameof(TokenSummaryEntity.MarketId)} = 0)";
+        var filterOnMarket = request.MarketId > 0;
+
+        var whereFilterBuilder = new StringBuilder();
+        if (filterOnMarket) whereFilterBuilder.Append($" WHERE ts.{nameof(TokenSummaryEntity.MarketId)} = @{nameof(SqlParams.MarketId)}");
```

Comment:
I think we have to keep the ternary logic and the initial join. Multiple markets can return multiple rows because of the join on `token_summary`. So if we filter for a specific market, it'll be fine, if not, then it can return the same token for each market its within.

Maybe its time to split this out into a MarketTokens vs Tokens filter

Reply:
I can do some testing around this, my belief is that the `GROUP BY` will limit it to return only a single row for each token even with multiple markets.

Reply:
I appear to have got it working by defining a one to one relationship using dapper multi-mapping. Taking the following data:
![image](https://user-images.githubusercontent.com/39023939/147982281-7d184aac-0115-48c4-a249-f9aa7b69a6a3.png)

Call to `GET /tokens` returns
```json
{
    "address": "PRpQxG3nFPzs4KZfKgFYteYkoBHGQKUH1V",
    "name": "Bitcoin (Wrapped)",
    "symbol": "xBTC",
    "decimals": 8,
    "sats": 100000000,
    "totalSupply": "21000000.00000000",
    "summary": {
        "priceUsd": 528.411446000000,
        "dailyPriceChangePercent": -1.430000000000,
        "modifiedBlock": 1472
    }
}
```

Call to `GET /markets/PQKXghAygiEGfE7TaLymHKGiZ42nG7DvrM/tokens` returns
```json
{
    "market": "PQKXghAygiEGfE7TaLymHKGiZ42nG7DvrM",
    "liquidityPool": "PKQc4Lj7pTy9se1B3qBLDQv33nZ34bnSP1",
    "address": "PRpQxG3nFPzs4KZfKgFYteYkoBHGQKUH1V",
    "name": "Bitcoin (Wrapped)",
    "symbol": "xBTC",
    "decimals": 8,
    "sats": 100000000,
    "totalSupply": "21000000.00000000",
    "summary": {
        "priceUsd": 536.290000000000,
        "dailyPriceChangePercent": 1.430000000000,
        "modifiedBlock": 1428
    }
}
```

Splitting them into two handlers can still be done, though probably isn't necessary unless there's a strong reason to optimise them.

Reply:
Ok thats nice if it works! Only suggestion is to wrap in `ROUND(AVG(...), 8 )` so our percent and usd decimals stick to 8 decimals.

Reply:
Added ROUND